src/test/bin/winkernel/control.cpp

```diff
@@ -81,7 +170,20 @@ QuicTestCtlInitialize(
     WDF_IO_QUEUE_CONFIG QueueConfig;
     WDFQUEUE Queue;
 
-    MsQuic = new (std::nothrow) MsQuicApi();
+    Status = QuicTestRegisterNmrClient();
+    if (!NT_SUCCESS(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "QuicTestRegisterNmrClient failed");
+        goto Error;
+    }
+
+    MsQuic =
```

Comment:
Verify the ProviderDispatch version first? Though I suppose since its version is currently set to zero, I think your check would be impossible to fail 😄

Reply:
I am not exactly sure how to use all the version fields NMR. There's also a version field in the NPI_REGISTRATION_INSTANCE passed in the attach callback. MsQuic's entry function also has an argument for version.

Would it make sense to just defer the version related stuff to MsQuicOpenVersion and assume there are always and only MsQuicOpenVersion and MsQuicClose in the provider dispatch?

Reply:
(the NMR doc just says for network modules set version to 0??)